requests/cookies.py

```diff
@@ -123,8 +123,8 @@ def extract_cookies_to_jar(jar, request, response):
     :param request: our own requests.Request object
     :param response: urllib3.HTTPResponse object
     """
-    if not (hasattr(response, '_original_response') and
-            response._original_response):
+    if jar is None or not (hasattr(response, '_original_response') and
```

Comment:
For clarity's sake it'd be better to have this be two separate `if` statements, rather than one gigantic one that evaluates all possible early-exit conditions.